satpy/readers/viirs_l2.py

```diff
@@ -0,0 +1,188 @@
+        """Get platform name."""
+        try:
+            res = self.get("/attr/platform", self.filename_info["platform_shortname"])
+        except KeyError:
+            res = "Unknown"
+
+        return {
+            "JPSS-1": "NOAA-20",
+            "NP": "Suomi-NPP",
+            "J1": "NOAA-20",
+            "J2": "NOAA-21",
+            "JPSS-2": "NOAA-21",
+        }.get(res, res)
+
+    @property
+    def sensor_name(self):
+        """Get sensor name."""
+        return self["/attr/instrument"].lower()
+
+    def _get_dataset_file_units(self, ds_info, var_path):
+        file_units = ds_info.get("units")
+        if file_units is None:
+            file_units = self.get(var_path + "/attr/units")
+        if file_units == "none" or file_units == "None":
+            file_units = "1"
+        return file_units
+
+    def _get_dataset_valid_range(self, ds_info, var_path):
+        valid_min = self.get(var_path + "/attr/valid_min")
+        valid_max = self.get(var_path + "/attr/valid_max")
+        if not valid_min and not valid_max:
+            valid_range = self.get(var_path + "/attr/valid_range")
+            if valid_range is not None:
+                valid_min = valid_range[0]
+                valid_max = valid_range[1]
+        scale_factor = self.get(var_path + "/attr/scale_factor")
+        scale_offset = self.get(var_path + "/attr/add_offset")
+        return valid_min, valid_max, scale_factor, scale_offset
+
+    def get_metadata(self, dataset_id, ds_info):
+        """Get metadata."""
+        var_path = ds_info.get("file_key", ds_info["name"])
+        file_units = self._get_dataset_file_units(ds_info, var_path)
+
+        # Get extra metadata
+        i = getattr(self[var_path], "attrs", {})
+        i.update(ds_info)
+        i.update(dataset_id.to_dict())
+        i.update(
+            {
+                "file_units": file_units,
+                "platform_name": self.platform_name,
+                "sensor": self.sensor_name,
+                "start_orbit": self.start_orbit_number,
+                "end_orbit": self.end_orbit_number,
+            }
+        )
+        i.update(dataset_id.to_dict())
+        return i
+
+    def adjust_scaling_factors(self, factors, file_units, output_units):
+        """Adjust scaling factors."""
+        if factors is None or factors[0] is None:
+            factors = [1, 0]
+        if file_units == output_units:
+            LOG.debug("File units and output units are the same (%s)", file_units)
+            return factors
+        factors = np.array(factors)
+
+        if file_units == "1" and output_units == "%":
+            LOG.debug(
+                "Adjusting scaling factors to convert '%s' to '%s'",
+                file_units,
+                output_units,
+            )
+            factors[::2] = np.where(factors[::2] != -999, factors[::2] * 100.0, -999)
+            factors[1::2] = np.where(factors[1::2] != -999, factors[1::2] * 100.0, -999)
+            return factors
```

Comment:
If this isn't used, this could be removed. Are there any level 2 products that are the same as the Level 1B datasets? I know for ABI there are L2 products that are the same as the calibrated L1 band data.